### PR TITLE
add spring CVE-2022-22965,CVE-2022-22978

### PR DESCRIPTION
**Pull request by QdghJ**
add spring CVE-2022-22965,CVE-2022-22978

### CODE REVIEW

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: "CVE-2022-22978 Spring Security Authorization Bypass"
+    description: "find CVE-2022-22978 Spring Security Authorization Bypass"
+    author: "timeshatter"
+    tags: "Spring Security Authorization Bypass","CVE-2022-22978"
+
+given request then
+    send request called check :
+        appending path: "%0a"
+
+    if {check.response.status_code} is "200" and not({base.response.status_code} is "200") then
```
Comment (PortSwiggerWiener) [inline vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:12]: Can we check the base response status code before issuing the request?

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      report issue:
+        severity: high
```
Comment (PortSwiggerWiener) [inline vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:14]: Please can you fix the indentation.

File: vulnerabilities-CVEd/CVE-2022-22965-spring_data_binding_rce.bcheck [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    name: "CVE-2022-22965 Spring Data Binding RCE"
+    description: "find CVE-2022-22965 Spring Data Binding RCE"
+    author: "timeshatter"
+    tags: "Spring Data Binding RCE","CVE-2022-22965"
+
+define:
+    poc = `class.module.classLoader.URLs%5B0%5D=0`
+
+given request then
+    send request called check :
```
Comment (PortSwiggerWiener) [inline vulnerabilities-CVEd/CVE-2022-22965-spring_data_binding_rce.bcheck:12]: Can we check the base response before issuing the request?

Also could we include an additional 'control' payload to detect if an app is simple returning a 400 when it receives an unexpected parameter? E.g. we could check the base response, issue your payload, then issuing the control which is similar but not semantically meaningful and only report if the latter does not also return a 400.

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        confidence: firm
```
Comment (PortSwiggerWiener) [inline vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:15]: Do you think 'firm' is the correct confidence here? We've not really fingerprinted that the app is using Spring.

**Review by PortSwiggerWiener — COMMENTED** [review]
Many thanks for the submission and the interesting BChecks!

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck (hunk 1 above)
Comment (QdghJ) [reply vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:12]: You are right. I didn't think of that. Thank you.

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck (hunk 2 above)
Comment (QdghJ) [reply vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:14]: Okay

File: vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck (hunk 4 above)
Comment (QdghJ) [reply vulnerabilities-CVEd/CVE-2022-22978-spring_security_auth_bypass.bcheck:15]: At present, I don't have a good way to identify spring, so let's set it as tentative

File: vulnerabilities-CVEd/CVE-2022-22965-spring_data_binding_rce.bcheck (hunk 3 above)
Comment (QdghJ) [reply vulnerabilities-CVEd/CVE-2022-22965-spring_data_binding_rce.bcheck:12]: Your idea is great, thank you. I will make the following changes

**Pull request merged**